osquery/tables/system/darwin/processes.cpp

```diff
@@ -266,6 +267,68 @@ void genProcNumThreads(QueryContext& context, int pid, Row& r) {
   }
 }
 
+uint64_t parseKCItem(kcdata_item_t item) {
+  if (item->type == STACKSHOT_KCTYPE_TASK_SNAPSHOT) {
+    struct task_snapshot_v2* tsv2 = reinterpret_cast<struct task_snapshot_v2*>(item->data);
+    return tsv2->ts_unique_pid;
+  } else {
+    return std::numeric_limits<std::uint64_t>::max();
+  }
+}
+
+void genProcUniquePid(QueryContext& context, int pid, Row& r) {
+  if (!context.isColumnUsed("unique_pid")) {
+    return;
+  }
+
+  // A definition of the stackshot_config structure so we can pull data out of
+  // the kernel
+  typedef struct stackshot_config {
+    int pid;
+    uint32_t flags;
+    uint64_t dtimestamp;
+    uint64_t buf;
+    uint32_t buf_size;
+    uint64_t out_buf_addr;
+    uint64_t out_buf_size_addr;
+  } stackshot_config_t;
+
+  char* addr = nullptr;
+  uint64_t size = 0;
+
+  stackshot_config_t stconf = {0};
+  stconf.pid = pid;
+  stconf.buf = 0;
+  stconf.buf_size = 0;
+  stconf.flags = 0x10000; // STACKSHOT_KCDATA_FORMAT
+  stconf.out_buf_addr = reinterpret_cast<uint64_t>(&addr);
+  stconf.out_buf_size_addr = reinterpret_cast<uint64_t>(&size);
+
+  // Syscall 491 is the syscall to get stackshots which can return data from
+  // the stack of all running process or only one if a pid is provided
+  auto rc = syscall(491, 1, (uint64_t)&stconf, sizeof(stconf));
```

Comment:
Do the syscalls have names? If so can we put 491 as a constant?

Reply:
Use `reinterpret_cast`.

Reply:
Syscalls do have names but this one is private. Since we only use it once, I'm reluctant to put it as a constant. Disagree?